src/mpl/src/gavl/mpl_gavl.c

```diff
@@ -38,6 +38,11 @@ typedef struct gavl_tree_node {
     const void *val;
 } gavl_tree_node_s;
 
+typedef struct gavl_tree_removed_node {
+    gavl_tree_node_s *node;
```

Comment:
Each `gavl_tree_node_s` already have 3 pointers, and we `malloc` a new structure just because we want a pointer named `next`? I find this a bit silly. How about make `gavl_tree_node_s` a union, then you'll have your `next` pointer with the right name

Reply:
sounds good, I will fix it.